src/utils/explorer.ts

```diff
@@ -0,0 +1,33 @@
+import { NetworkNames } from '../constants';
+import { detectProvider, MarinaProvider } from 'marina-provider';
+
+export const Explorers = {
+  [NetworkNames.MAINNET]: 'https://blockstream.info/liquid/api',
+  [NetworkNames.TESTNET]: 'https://blockstream.info/liquidtestnet/api',
+}
+
+/**
+ * Returns the explorer to use for a given network
+ * @param network
+ * @returns url
+ */
+export function getExplorerForNetwork(network: string): string {
+  return Explorers[network] || Explorers[NetworkNames.MAINNET];
+}
+
+/**
+ * Checks for network selected on Marina and returns explorer for that network
+ * @returns url
+ */
+export async function getExplorer(): Promise<string> {
+  const network = await detectProvider('marina')
+    .then(async (marina: MarinaProvider): Promise<string> => {
+      const network = await marina.getNetwork();
```

Comment:
You don't need to refetch the network from Marina each time u need it. `$marinastore` from ConnectButton is using the marina event `NETWORK` to auto-update `$marinastore.network`

Reply:
I'm new to Svelte, and was not able to access $marinaStore from a .ts file, only from .svelte files. Can you help me out here on how to do it? Thanks in advance.

Reply:
To be honest, I don't know (dex.vulpem is also my first time with svelte) but I would suggest you to add a parameter `network` in the `getExplorer` function and then use `getExplorer($marinastore.network)` in Svelte components or something similar

Reply:
Yes stores are only available in svelte files. But regardless, looks best to me to keep pure .ts utils not aware of underlying UI framework used, so yes agree in having the network a parameter here

Reply:
I have that function already, it's called `getExplorerForNetwork(network)`.

This `getExplorer()` is needed inside `createCoinStore()`, which is then used like this:
```
export const coinStore = createCoinStore(ASSETS);
```

Not sure how I can pass `$marinastore.network` as a parameter to the `coinStore` import without changing a lot of code. Thought a lot about this, and this seemed the best solution. But I'm always ready to learn.

Reply:
`updateWithAssets` should have a parameter for the explore? so we uplift the `getExploer` in the smart/parent component which is connected to the network

Reply:
Commit c394109e should fix this.